subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/BuildLifecycleAwareVirtualFileSystem.java

```diff
@@ -43,5 +43,5 @@
     /**
      * Called when the build is finished.
      */
-    void beforeBuildFinished(boolean watchingEnabled, boolean verboseLogging, BuildOperationRunner buildOperationRunner, int maximumNumberOfWatchedHierarchies);
+    void beforeBuildFinished(boolean watchingEnabled, boolean verboseVfsLogging, boolean debugWatchLogging, BuildOperationRunner buildOperationRunner, int maximumNumberOfWatchedHierarchies);
```

Comment:
Shall we use enums or a configuration object instead of passing in three `boolean`s here?

Reply:
I've added  enums